### PR TITLE
[MIG] product_warranty: Migration to 16.0

### PR DESCRIPTION
**Pull request by imarulbala**
[MIG] product_warranty: Migration to 16.0

### CODE REVIEW

**Comment by rousseldenis** [comment]
@imarulbala Don't integrate flake8 fix here.

I'll do it

**Comment by rousseldenis** [comment]
@imarulbala Could you rebase ?

**Comment by StefanRijnhart** [comment]
Please add the module name to the pre-commit commit description:
image

and move the version number update to the migration commit.

File: product_warranty/views/product_warranty.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    <record id="product_return_instructions_action" model="ir.actions.act_window">
+        <field name="name">Products Return Instructions</field>
+        <field name="res_model">return.instruction</field>
+        <field name="view_mode">tree,form</field>
+        <field name="view_id" ref="product_return_instructions_tree_view" />
+    </record>
+    <menuitem
+        action="product_return_instructions_action"
+        groups="base.group_no_one"
+        id="menu_product_return_instructions_action"
```
Comment (StefanRijnhart) [inline product_warranty/views/product_warranty.xml]: Please move the instructions to their own `views/return_instructions_views.xml` file.

File: product_warranty/views/product_warranty.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        parent="sale.prod_config_main"
+        sequence="3"
+    />
+    <!-- supplier info tree view -->
+    <record id="product_supplierinfo_warranty_tree_view" model="ir.ui.view">
+        <field name="name">product.supplierinfo.warranty.tree</field>
+        <field name="model">product.supplierinfo</field>
```
Comment (StefanRijnhart) [inline product_warranty/views/product_warranty.xml]: Please move to a file `views/product_supplierinfo_views.xml`.

File: product_warranty/README.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+=====
+
+To use this module, you need to:
+
+#. Go to *Sales > Products > Products (or Product Variants)*
+#. Create a new product (or product variant) or edit an existing one
+   and set 'Warranty Duration' under 'Sales' tab.
+#. If 'Purchase' module is installed, got to
+   *Sales > Products > Products (or Product Variants)*, go to 'Purchase' tab,
+   edit supplier information lines an set the warranty information for each one.
```
Comment (StefanRijnhart) [inline product_warranty/README.rst:52]: Can you add a line like this in a file called `readme/CONFIGURATION.rst` with the following contents:

```
To manage the available return instructions, enable debug mode and go to 
*Sales -> Configuration -> Product Return Instructions*.
```

File: product_warranty/views/product_warranty.xml (hunk 1 above)
Comment (dynapps) [reply product_warranty/views/product_warranty.xml]: Done!

File: product_warranty/views/product_warranty.xml (hunk 2 above)
Comment (dynapps) [reply product_warranty/views/product_warranty.xml]: Done!

File: product_warranty/README.rst (hunk 3 above)
Comment (dynapps) [reply product_warranty/README.rst:52]: Added to `readme/CONFIGURE.rst` ;-)

Comment (StefanRijnhart) [reply product_warranty/README.rst:52]: Much better than my suggestion!

**Review by StefanRijnhart — APPROVED** [review]
Thanks! Code reviewed + tested.

File: product_warranty/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+"proveedor y marca del fabricante. No significa necesariamente que la "
+"garantía que se aplica es el de la empresa que hace la devolución (es decir: "
+"se puede devolver a la empresa y estar bajo la garantía de la marca)"
+
+#. module: product_warranty
+#: model:ir.model.fields.selection,name:product_warranty.selection__product_template__warranty_type__year
+msgid "Year(s)"
+msgstr ""
+
+#~ msgid ""
```
Comment (rousseldenis) [inline product_warranty/i18n/es.po:279]: Could you remove unused code ?

File: product_warranty/models/product_supplierinfo.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                partner_id = record.company_id.crm_return_address_id.id
+            elif (
+                return_partner
+                and return_partner == "other"
+                and record.warranty_return_other_address
+            ):
+                partner_id = record.warranty_return_other_address.id
+            record.warranty_return_address = partner_id
+
+    warranty_duration = fields.Float(
```
Comment (rousseldenis) [inline product_warranty/models/product_supplierinfo.py:43]: Could you put fields on top ?

File: product_warranty/tests/test_product_warranty.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Copyright 2016 Cyril Gaudin (Camptocamp)
+# Copyright 2015 Vauxoo, Yanina Aular
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductWarranty(TransactionCase):
+    def setUp(self):
+        super(TestProductWarranty, self).setUp()
```
Comment (rousseldenis) [inline product_warranty/tests/test_product_warranty.py:10]: Could you switch to classmethod ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by pedrobaeza** [comment]
Superseded by #358

**Pull request closed without merging**